plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/action/CompareWithLatestRepositoryAction.java

```diff
@@ -0,0 +1,116 @@
+import org.eclipse.che.api.workspace.shared.dto.ProjectConfigDto;
+import org.eclipse.che.ide.api.action.ActionEvent;
+import org.eclipse.che.ide.api.app.AppContext;
+import org.eclipse.che.ide.api.notification.NotificationManager;
+import org.eclipse.che.ide.api.project.node.HasStorablePath;
+import org.eclipse.che.ide.api.selection.Selection;
+import org.eclipse.che.ide.api.selection.SelectionAgent;
+import org.eclipse.che.ide.ext.git.client.GitLocalizationConstant;
+import org.eclipse.che.ide.ext.git.client.compare.ComparePresenter;
+import org.eclipse.che.ide.ext.git.client.compare.changedList.ChangedListPresenter;
+import org.eclipse.che.ide.part.explorer.project.ProjectExplorerPresenter;
+import org.eclipse.che.ide.rest.AsyncRequestCallback;
+import org.eclipse.che.ide.rest.StringUnmarshaller;
+import org.eclipse.che.ide.ui.dialogs.ConfirmCallback;
+import org.eclipse.che.ide.ui.dialogs.DialogFactory;
+
+import java.util.Collections;
+
+import static org.eclipse.che.api.git.shared.DiffRequest.DiffType.NAME_STATUS;
+
+/** @author Igor Vinokur */
+@Singleton
+public class CompareWithLatestRepositoryAction extends GitAction {
+    private final ComparePresenter        comparePresenter;
+    private final ChangedListPresenter    changedListPresenter;
+    private final DialogFactory           dialogFactory;
+    private final NotificationManager     notificationManager;
+    private final GitServiceClient        gitService;
+    private final GitLocalizationConstant locale;
+    private final SelectionAgent          selectionAgent;
+
+    private final String REVISION = "HEAD";
+
+    @Inject
+    public CompareWithLatestRepositoryAction(ComparePresenter presenter,
+                         ChangedListPresenter changedListPresenter,
+                         AppContext appContext,
+                         DialogFactory dialogFactory,
+                         NotificationManager notificationManager,
+                         GitServiceClient gitService,
+                         GitLocalizationConstant constant,
+                         SelectionAgent selectionAgent,
+                         ProjectExplorerPresenter projectExplorer) {
+        super(constant.compareTitle(), constant.compareTitle(), null, appContext, projectExplorer);
+        this.comparePresenter = presenter;
+        this.changedListPresenter = changedListPresenter;
+        this.dialogFactory = dialogFactory;
+        this.notificationManager = notificationManager;
+        this.gitService = gitService;
+        this.locale = constant;
+        this.selectionAgent = selectionAgent;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        ProjectConfigDto project = appContext.getCurrentProject().getRootProject();
+        String pattern;
+        String path;
+
+        Selection<HasStorablePath> selection = (Selection<HasStorablePath>)selectionAgent.getSelection();
```

Comment:
Dont't do in this way. Selection agent returns Selection<?>  and you can catch class cast exception/